### PR TITLE
ML: Empty state is shown after deleting all assets from a page

### LINKED ISSUE #12648: ML: Empty state is shown after deleting all assets from a page
## Bug report

After all assets are deleted from a specific page in the media library, the ML shows an empty state (without pagination ...).

### Steps to reproduce the behavior

1. Go to 'ML'
2. Upload more than 10 assets
3. Go to page 2
4. Select & delete all assets

### Expected behavior

I'd expect the ML to go back to the previous page (if there is any), if it doesn't contain any assets after deletion.

### Screenshots

[If applicable, add screenshots to help explain your problem.](https://user-images.githubusercontent.com/2244375/155495740-21bfecb8-16a4-4056-a06f-08d536e80bac.mp4
)

### System

- Node.js version: 14
- NPM version: 6
- Strapi version: 4.1.0
- Database: sqlite
- Operating system: MacOS

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #12648

### What does it do?

- After deleting all assets from a page that is not the first page, a user is routed to the previous page that still has assets

### Why is it needed?

- After deleting all assets from a page that is not the first page, a user stayed on  that page showing an empty page without assets even when the previous page has assets.

### How to test it?

1. From the menu navigate to the Media Library
2. Add enough assets to be displayed on 2 pages.
3. Navigate to the second page and delete all its assets
    - After successful deletion, you should be routed to page 1 that still has assets

#### Demo
https://user-images.githubusercontent.com/45232708/205258689-04da6210-440c-41dc-b714-ae8341dbdfa0.mov

### CODE REVIEW

**Comment by gu-stav** [comment]
@gitstart Thank you for this PR! Could you fix the eslint error, before I start a review?

**Comment by gu-stav** [comment]
@gitstart Thank you very much. I can see one failing unit/ integration test now. Could you also fix that one and add a new one for the enhancement you added?

**Comment by gu-stav** [comment]
@gitstart are you still working on the PR?

**Comment by arunjagadishsd** [comment]
> @gitstart are you still working on the PR?

Yes pushed new code

**Comment by gitstart** [comment]
> @gitstart are you still working on the PR?

Yes pushed the latest code

**Comment by gu-stav** [comment]
@gitstart Something went wrong with the latest change - it looks as if you pulled in an old state of the `main` branch, which leads to more than 270 files being reported as changed.

Could you update your fork to upstream before rebasing/ merging main, please? 🙏🏼

File: packages/core/upload/admin/src/pages/App/MediaLibrary.js [hunk 1]
```diff
@@ -127,6 +127,15 @@ export const MediaLibrary = () => {
     setShowEditFolderDialog((prev) => !prev);
   };
 
+  useEffect(() => {
+    if (assetCount > 0) {
+      setQuery({
+        ...query,
+        assetCount,
```
Comment (gu-stav) [inline packages/core/upload/admin/src/pages/App/MediaLibrary.js:134]: Mh, I'm not a fan of this way to pass around state tbh. I can think of two ways to solve that problem:

1. Call a callback after `handleConfirmRemove()` was called and then do the redirection. In `MediaLibrary` you do have the total asset count available.
2. Pass down the total as a prop to `BulkDeleteButton`

I think I'd prefer the first one. It doesn't hide the redirect in the delete button ... Wdyt?

Comment (gitstart) [reply packages/core/upload/admin/src/pages/App/MediaLibrary.js:134]: Yeah I think I would try to implement it in the first way but I think we'd also need to change the tests too

**Comment by gitstart** [comment]
Hi @gu-stav,
The implementation logic has been updated, as recommended.
Here's an updated version of the demo video, could you please take a look at it?.
Thanks.
https://user-images.githubusercontent.com/45232708/205258689-04da6210-440c-41dc-b714-ae8341dbdfa0.mov

**Comment by gitstart** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @PrincesaTyne, @raph941, @phunguyenmurcul.

**Pull request merged**